src/Controls/src/Core/Compatibility/Handlers/ListView/iOS/ListViewRenderer.cs

```diff
@@ -306,10 +307,17 @@ public override void TraitCollectionDidChange(UITraitCollection previousTraitCol
 			base.TraitCollectionDidChange(previousTraitCollection);
 #pragma warning restore CA1422 // Validate platform compatibility
 			// Make sure the cells adhere to changes UI theme
-			if (OperatingSystem.IsIOSVersionAtLeast(13) && previousTraitCollection?.UserInterfaceStyle != TraitCollection.UserInterfaceStyle)
+			if (OperatingSystem.IsIOSVersionAtLeast(13) && !UIStyleMatches(previousTraitCollection))
 				ReloadData();
 		}
 
+		[SupportedOSPlatform("ios13.0")]
+		[SupportedOSPlatform("maccatalyst13.1")]
+		bool UIStyleMatches (UITraitCollection target)
+		{
+			return target.UserInterfaceStyle == TraitCollection.UserInterfaceStyle;
```

Comment:
Are these nullchecks (previousTraitCollection?.UserInterfaceStyle) removed on purpose?

Also I have found that there is method "UITraitCollection.HasDifferentColorAppearanceComparedTo(UITraitCollection previous)" but have not tried. 
Maybe this could be cheaper (passing C# referenced object to exported method vs. accessing and comparing two obj-c objects in C# referenced objects and comparing), but I am not sure how it handles "Unspecified" one or null.